### PR TITLE
Improve error message in Rasa Transformer - [ENG46]

### PR DESCRIPTION
**Pull request by varunshankar**
**Proposed changes**:
- Exit the training upon checking the transformer_size with a meaningful error message.
- Update the information in Rasa docs.
- Add tests for the change.
   Fixes https://rasahq.atlassian.net/browse/ENG-46

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [x] updated the documentation
- [x] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/main/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: docs/docs/components.mdx [hunk 1]
```diff
@@ -1394,6 +1394,8 @@ More details on the parameters can be found on the [scikit-learn documentation p
   * `transformer_size`:
     This parameter sets the number of units in the transformer (default: `256`).
     The vectors coming out of the transformers will have the given `transformer_size`.
+    The `transformer_size` should be a multiple of the `number_of_attention_heads` parameter,
+    the training exists with a Value Error otherwise.
```
Comment (twerkmeister) [inline docs/docs/components.mdx:1398]:
```suggestion
    the training exits with an error otherwise.
```

File: docs/docs/components.mdx [hunk 2]
```diff
@@ -2342,6 +2344,8 @@ The `SpacyEntityExtractor` extractor does not provide a `confidence` level and w
     * `transformer_size`:
       This parameter sets the number of units in the transformer (default: `256`).
       The vectors coming out of the transformers will have the given `transformer_size`.
+      The `transformer_size` should be a multiple of the `number_of_attention_heads` parameter,
+      the training exists with a Value Error otherwise.
```
Comment (twerkmeister) [inline docs/docs/components.mdx:2348]:
```suggestion
      the training exits with an error otherwise.
```

File: docs/docs/components.mdx [hunk 3]
```diff
@@ -2679,6 +2683,8 @@ Selectors predict a bot response from a set of candidate responses.
   * `transformer_size`:
     This parameter sets the number of units in the transformer (default: `None`).
     The vectors coming out of the transformers will have the given `transformer_size`.
+    The `transformer_size` should be a multiple of the `number_of_attention_heads` parameter,
+    the training exists with a Value Error otherwise.
```
Comment (twerkmeister) [inline docs/docs/components.mdx:2687]:
```suggestion
    the training exits with an error otherwise.
```

**Review by twerkmeister — APPROVED** [review]
Good detective work tracking down earlier work on this topic. Spotted some typos and suggested not mentioning error type explicitly.

**Comment by varunshankar** [comment]
> Good detective work tracking down earlier work on this topic. Spotted some typos and suggested not mentioning error type explicitly.

Thank you for taking the time to help with the code review. And thanks for catching the typos.

**Pull request merged**